cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -149,7 +183,23 @@ class knn_callback {
             const auto ndeps = deps + copy_indices + copy_distances;
             auto temp_resp = temp_resp_.get_row_slice(0, to - from);
             auto s_event = select_indexed(queue_, inp_indices, inp_responses_, temp_resp, ndeps);
-            comp_responses = voting_->operator()(temp_resp, out_block, { s_event });
+
+            if (uniform_voting_) {
+                comp_responses = uniform_voting_->operator()(temp_resp, out_block, { s_event });
```

Comment:
Why not simply
```suggestion
                comp_responses = uniform_voting_(temp_resp, out_block, { s_event });
```

Reply:
It's incorrect. `uniform_voting_` is a pointer

Reply:
Discussed